ayon_api/entity_hub.py

```diff
@@ -51,8 +79,11 @@ class EntityHub(object):
         connection (ServerAPI): Connection to server with logged user.
 
     """
-
-    def __init__(self, project_name, connection=None):
+    def __init__(
+        self,
+        project_name: str,
+        connection: Optional[ServerAPI] = None
+    ) -> None:
         if not connection:
```

Comment:
Even though this is valid, MyPy is fighting a little bit with possibility that `connection` can be `None` after all. This little change will help it a lot:

```suggestion
        if connection is None:
```